src/AlgebraicGeometry/Schemes/AffineSchemes/Objects/Attributes.jl

```diff
@@ -480,6 +480,10 @@ julia> codim(Y)
   return dim(ideal(ambient_coordinate_ring(X), [zero(ambient_coordinate_ring(X))])) - dim(X)
 end
 
+@attr Int function degree(X::Spec{BRT, RT}; check::Bool=true) where {BRT<:Field, RT}
+  @check dim(X) == 0 "The affine scheme X needs to be zero-dimensional"
+  return vector_space_dimension(OO(X))
+end
 
```

Comment:
So for now, I would keep the check but remove `@attr` even if the output of `degree` is not cached then.
```suggestion
function degree(X::Spec{BRT, RT}; check::Bool=true) where {BRT<:Field, RT}
  @check dim(X) == 0 "Degree is not implemented for affine schemes of dimension > 0"
  return vector_space_dimension(OO(X))
end
```